src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6595,3 +6600,30 @@ void os::print_memory_mappings(char* addr, size_t bytes, outputStream* st) {
     st->cr();
   }
 }
+
+bool os::read_bootid(char *dest, size_t size) {
+  int fd = ::open("/proc/sys/kernel/random/boot_id", O_RDONLY);
```

Comment:
Bood_id looks interesting! But AFAICS it remains the same for each new container, and the boot time may have been adjusted for that container.
```
anton@mercury:~$ docker run -it ubuntu:20.04 cat /proc/sys/kernel/random/boot_id
9b913973-3082-471a-add5-6b802a04a9b2
anton@mercury:~$ docker run -it ubuntu:20.04 cat /proc/sys/kernel/random/boot_id
9b913973-3082-471a-add5-6b802a04a9b2
anton@mercury:~$ cat /proc/sys/kernel/random/boot_id
9b913973-3082-471a-add5-6b802a04a9b2
```

Should not we mix something extra to boot_id, for example, a hostname (which is different for each container)?

Reply:
The aim of this work is to prevent 'random' readings. New containers will keep the `boot_id` but their monotonic time *might* be adjusted, true. I agree with the comment above that in that case we should still ensure monotonicity, but since the readings are not 'random' I think that we can assume that the user did adjust monotonic time intentionally, and the application should observe that.

Reply:
OK, agree. This may be a comment in the code, explaining that we leave some control of the time in the user's hands.